### ISSUE TITLE
assert is not defined as a test variable for Mocha

### ISSUE THREAD

**Issue by Treeek** [eslint/eslint#11150]
**Tell us about your environment**

* **ESLint Version:** v5.9.0
* **Node Version:** v8.11.4
* **npm Version:** v6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**

<details>
<summary>Configuration
</summary>

```json
{
    "env": {
        "es6": true,
        "node": true,
        "mocha":true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "ecmaVersion": 2018,
        "sourceType": "module"
    },
    "rules": {
        "no-console": "off",
        "indent": [
            "error",
            "tab"
        ],
        "linebreak-style": [
            "error",
            "windows"
        ],
        "quotes": [
            "error",
            "double"
        ],
        "semi": [
            "error",
            "always"
        ]
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
describe("Just Sum test", () => {
	it("adds two numbers", (done) => {
		assert(17 + 1 === 18);
		done();
	});
});

```

```bash
eslint ../test/file.js
```

**What did you expect to happen?**
Since I defined mocha as a environment in my .eslintrc file, I expected to have all test global variables defined.

**What actually happened? Please include the actual, raw output from ESLint.**
7:4 error  'assert' is not defined  no-undef

**Are you willing to submit a pull request to fix this bug?**
Yes

**Comment by not-an-aardvark** [comment]
Does Mocha actually define an `assert` global variable? I was under the impression that you're supposed to provide your own assertion library when using Mocha.

**Comment by Treeek** [comment]
wow @not-an-aardvark you are right, I forgot that nodejs had its  own assert(). 
My bad, my bad.
My fix for this error was: 
```js
describe("Just Sum test", () => {
	it("adds two numbers", (done) => {
                const assert = require("assert");
		assert(17 + 1 === 18);
		done();
	});
});
```
